From b520385be65f2eb6c72475d846448cfab0c46b3b Mon Sep 17 00:00:00 2001
From: Francois Gaffie <francois.gaffie@intel.com>
Date: Tue, 30 Apr 2013 16:56:34 +0200
Subject: [PORT FROM R4.1] Lost of 2 seconds of call when accepting it

BZ: 29679

The setMode in call was recently changed and is sent earlier.
The sequence is:
 - setMode In call from the policy
 - ModemCallStatus=true received from AT Manager
 - setStreamParameter with the device to use received from the policy.

As a result, when receiving the call status, the route is established
on the wrong device. When the setStreamParameter is received, a re-
routing is performed, increasing by around 800ms the establishment
of the audio call.

This patch ensures the policy always call setMode followed by
setStreamParameter on the output device.

Feature: AudioComms
Category: Integration
Purpose: PV-Enabler
Origin: Internal
Upstream-candidate: no
Change-Id: I44c774008f841e414e5025fba29ae1a0e24f7f3d
Orig-Change-Id: I596a4aa6971b570d352586060a67830dcd834167
Signed-off-by: Francois Gaffie <francois.gaffie@intel.com>
---
 audio/AudioPolicyManagerBase.cpp |   21 +--------------------
 1 files changed, 1 insertions(+), 20 deletions(-)

diff --git a/audio/AudioPolicyManagerBase.cpp b/audio/AudioPolicyManagerBase.cpp
index 76f2240..da2346c 100644
--- a/audio/AudioPolicyManagerBase.cpp
+++ b/audio/AudioPolicyManagerBase.cpp
@@ -318,25 +318,6 @@ void AudioPolicyManagerBase::setPhoneState(int state)
     // store previous phone state for management of sonification strategy below
     int oldState = mPhoneState;
     mPhoneState = state;
-    bool force = false;
-
-    // are we entering or starting a call
-    if (!isStateInCall(oldState) && isStateInCall(state)) {
-        ALOGV("  Entering call in setPhoneState()");
-        // force routing command to audio hardware when starting a call
-        // even if no device change is needed
-        force = true;
-    } else if (isStateInCall(oldState) && !isStateInCall(state)) {
-        ALOGV("  Exiting call in setPhoneState()");
-        // force routing command to audio hardware when exiting a call
-        // even if no device change is needed
-        force = true;
-    } else if (isStateInCall(state) && (state != oldState)) {
-        ALOGV("  Switching between telephony and VoIP in setPhoneState()");
-        // force routing command to audio hardware when switching between telephony and VoIP
-        // even if no device change is needed
-        force = true;
-    }
 
     // check for device and output changes triggered by new phone state
     newDevice = getNewDevice(mPrimaryOutput, false /*fromCache*/);
@@ -379,7 +360,7 @@ void AudioPolicyManagerBase::setPhoneState(int state)
     }
 
     // change routing is necessary
-    setOutputDevice(mPrimaryOutput, newDevice, force, delayMs);
+    setOutputDevice(mPrimaryOutput, newDevice, true, delayMs);
 
     // if entering in call state, handle special case of active streams
     // pertaining to sonification strategy see handleIncallSonification()
-- 
1.7.4.1

